<?xml version="1.0"?>
<launch>
	<!-- arguments -->
	<arg name="domain_path" default="$(find assignment2_exprob)/planning/domain.pddl" />
	<arg name="problem_path" default="$(find assignment2_exprob)/planning/problem.pddl" />
	<!-- knowledge base -->
	<node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false"
output="screen">
	<param name="domain_path" value="$(arg domain_path)" />
	<param name="problem_path" value="$(arg problem_path)" />
	<!-- conditional planning flags -->
	<param name="use_unknowns" value="false" />
	</node>
	
	<!-- problem generation -->
	<include file="$(find rosplan_planning_system)/launch/includes/problem_interface.launch">
		<arg name="knowledge_base" value="rosplan_knowledge_base" />
		<arg name="domain_path" value="$(arg domain_path)" />
		<arg name="problem_path" value="$(find assignment2_exprob)/planning/new_problem.pddl" />
		<arg name="problem_topic" value="problem_instance" />
	</include>
	
	<include file="$(find rosplan_planning_system)/launch/includes/planner_interface.launch">
		<arg name="use_problem_topic" value="true" />
		<arg name="problem_topic" value="/rosplan_problem_interface/problem_instance" />
		<arg name="planner_topic" value="planner_output" />
		<arg name="domain_path" value="$(arg domain_path)" />
		<arg name="data_path" value="$(find assignment2_exprob)/planning/" />
		<arg name="planner_command" value="timeout 10 $(find rosplan_planning_system)/common/bin/popf DOMAIN PROBLEM" />
		
	</include>
  
  <!-- Parsing Interface -->
  <node name="rosplan_parsing_interface" pkg="rosplan_planning_system" type="pddl_simple_plan_parser" respawn="false" output="screen">
    <param name="knowledge_base" value="rosplan_knowledge_base" />
    <param name="planner_topic" value="/rosplan_planner_interface/planner_output" />
    <param name="plan_topic" value="complete_plan" />
  </node>

  <!-- Plan Dispatcher -->
  <node name="rosplan_plan_dispatcher" pkg="rosplan_planning_system" type="pddl_simple_plan_dispatcher" respawn="false" output="screen">
    <param name="knowledge_base" value="rosplan_knowledge_base" />
    <param name="plan_topic" value="/rosplan_parsing_interface/complete_plan" />
    <param name="action_dispatch_topic" value="action_dispatch" />
    <param name="action_feedback_topic" value="action_feedback" />
  </node>



<!-- move_to Actions -->
  <node name="rosplan_interface_move_to" pkg="assignment2_exprob" type="myaction2" respawn="false" output="screen">
    <param name="knowledge_base" value="rosplan_knowledge_base" />
    <param name="pddl_action_name" value="move_to" />
    <param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
    <param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
  </node>
  
  
  <node name="rosplan_interface_detect_marker" pkg="assignment2_exprob" type="myaction2" respawn="false" output="screen">
    <param name="knowledge_base" value="rosplan_knowledge_base" />
    <param name="pddl_action_name" value="detect_marker" />
    <param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
    <param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
  </node>
  
  <node name="rosplan_interface_go_to_least_id" pkg="assignment2_exprob" type="myaction2" respawn="false" output="screen">
    <param name="knowledge_base" value="rosplan_knowledge_base" />
    <param name="pddl_action_name" value="go_to_least_id" />
    <param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
    <param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
  </node>

</launch>



